dali/operators/generic/erase/erase.cu

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+#include "dali/core/static_switch.h"
+#include "dali/kernels/erase/erase_gpu.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/kernel_params.h"
+#include "dali/operators/generic/erase/erase.h"
+#include "dali/operators/generic/erase/erase_utils.h"
+#include "dali/pipeline/data/views.h"
+
+#define ERASE_SUPPORTED_NDIMS (1, 2, 3, 4, 5)
+#define ERASE_SUPPORTED_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, \
+                               uint64_t, int64_t, float)
+namespace dali {
+
+namespace detail {
+
+template <int ndim>
+kernels::ibox<ndim> make_box(const TensorShape<ndim> &lcorner, const TensorShape<ndim> &shape) {
+  auto lc = kernels::to_ivec(lcorner);
+  return kernels::ibox<ndim>(lc, lc + kernels::to_ivec(shape));
+}
+
+template <int ndim, typename Storage>
+TensorListView<Storage, kernels::ibox<ndim>, 1>
+as_boxes(TensorListView<Storage, int32_t, 3> tlv) {
+  TensorListShape<1> new_shape(tlv.shape.size());
+  for (int i = 0; i < tlv.shape.size(); ++i)
+    new_shape.set_tensor_shape(i, {tlv.shape[i][0]});
+  using ptrs_t = kernels::ibox<ndim>* const*;
+  return {reinterpret_cast<ptrs_t>(tlv.data.data()), new_shape};
+}
+
+template <int ndim, typename Storage>
```

Comment:
I don't think this overload is necessary - a non-const TensorListView should convert implicitly to const one.